contracts/FountainV1.sol

```diff
@@ -454,8 +457,8 @@ contract FountainV1 {
         // No pending moneyPool found, clone the latest moneyPool
         moneyPoolId = _getLatestMoneyPoolId(who);
         MoneyPool storage moneyPool = _createMoneyPoolFromId(moneyPoolId);
-        moneyPools[moneyPoolId] = moneyPool;
```

Comment:
@tauren this here seemed off. `moneyPoolId` refers to the latest id for the provided who, not the id of the newly created mp.

Reply:
Good eye! There is a bug here, but let's discuss the right solution, as it may not be what you are suggesting.

The line above has actually already been executed within `_createMoneyPoolFromId`. See line 598: `MoneyPool storage moneyPool = moneyPools[moneyPoolCount];`. Also, assigning `latestMoneyPoolIds` shouldn't happen here outside of `_createMoneyPoolFromId`, decoupled from where the MP creation happens. 

I propose we remove the last three lines of this function in your PR and change line 456 to:

```
return  _createMoneyPoolFromId(moneyPoolId);
```

And we change the return type of `_createMoneyPoolFromId` to return the ID instead of the MP as well as it assigning the new ID to `latestMoneyPoolIds`. 

Putting ID into the struct may ultimately be worthwhile and helpful. But I'm not 100% convinced it is necessary. And it makes me slightly uneasy since it creates the potential for the ID in the struct to not match the array ID that points to the struct.

Thoughts?

Reply:
great points, i'm also in favor of not adding the id to the struct. I'm less partial to the solution, more wanted to give a fix a go without simply opening an issue. I'll update with what you've suggested.